app/src/main/java/com/example/trashminder/presentation/createdReminder/HomeFragment.kt

```diff
@@ -63,4 +87,72 @@ class HomeFragment : Fragment() {
             }
         }
     }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    @Composable
+    private fun CheckNotificationPermission(it: Reminder, context: Context) {
+
+        var hasNotificationPermission by remember {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                mutableStateOf(
+                    ContextCompat.checkSelfPermission(
+                        context,
+                        Manifest.permission.POST_NOTIFICATIONS
+                    ) == PackageManager.PERMISSION_GRANTED
+                )
+            } else {
+                mutableStateOf(true)
+            }
+        }
+
+        val launcher = rememberLauncherForActivityResult(
+            contract = ActivityResultContracts.RequestPermission(),
+            onResult = { isGranted ->
+                hasNotificationPermission = isGranted
+            }
+        )
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            SideEffect {
+                launcher.launch(Manifest.permission.POST_NOTIFICATIONS)
+            }
+        }
+
+        if (hasNotificationPermission) {
+            setNotifications(it, context)
+        }
+    }
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+fun setNotifications(reminder: Reminder, context: Context) {
+
+    val formatter = DateTimeFormatter.ofPattern(
+        "dd/MM/yyyy HH:mm",
+        Locale.ENGLISH
+    )
+    val df = SimpleDateFormat("dd/MM/yyyy HH:mm", Locale.ENGLISH)
+
+    val localDateForTime = LocalDateTime.parse(reminder.date, formatter)
+    val timeInMilliseconds =
+        localDateForTime.atOffset(ZoneOffset.UTC).toInstant().toEpochMilli()
+
+    val currentTime = df.format(Calendar.getInstance().time)
+    Log.d("Notifications", "date: ${reminder.date}, currentTime: $currentTime")
+    val localDateForCurrentTime = LocalDateTime.parse(currentTime, formatter)
+    val currentTimeInMills =
+        localDateForCurrentTime.atOffset(ZoneOffset.UTC).toInstant().toEpochMilli()
+
+    if (timeInMilliseconds > currentTimeInMills) {
```

Comment:
What is this check all about?

Reply:
by my logic, the user should not be able to set a reminder on a date older than the current date, therefore
neither the notification can't be set (I know that the user currently can select an older date, but that is something that I wanted to solve later)